bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java

```diff
@@ -225,6 +225,22 @@ public Collection<WorkItem> prRun(Path scratchPath) {
         var seedPath = bot.seedStorage().orElse(scratchPath.resolve("seeds"));
         var hostedRepositoryPool = new HostedRepositoryPool(seedPath);
 
+        if (bot.confOverrideRepository().isEmpty()) {
+            //check whether this repo contains .jcheck/conf, if not, let the pr author to notify the repo owner
+            try {
+                pr.repository().fileContents(".jcheck/conf", pr.targetRef());
```

Comment:
Instead of explicitly checking for `.jcheck/conf` here through a remote call, I think it would be better to react to if the `Optional<CensusInstance>` below comes back empty or not. If it's empty and `bot.confOverrideRepository().isEmpty()` then you can print a message.

Reply:
> Instead of explicitly checking for `.jcheck/conf` here through a remote call, I think it would be better to react to if the `Optional<CensusInstance>` below comes back empty or not. If it's empty and `bot.confOverrideRepository().isEmpty()` then you can print a message.

Yesterday I was wondering whether there is any possible that `Optional<CensusInstance>` is empty due to other errors. Today I read through the code again and confirm that Only `.jcheck/conf` invalid or missing will make `Optional<CensusInstance>` empty. Thx!

Reply:
Maybe it would be even better to throw an exception instead of returning Optional.empty(). We can't recover from it  in CheckWorkItem anyway, and with a typed exception we can communicate more exactly what went wrong, and we can catch exactly that exception type and write the message.

Reply:
> Maybe it would be even better to throw an exception instead of returning Optional.empty(). We can't recover from it in CheckWorkItem anyway, and with a typed exception we can communicate more exactly what went wrong, and we can catch exactly that exception type and write the message.

I thought about it in the very beginning and I found that  if we choose to throw an exception, we will need to surround a lot of code with try catch because the scope of `census` is big.

Reply:
Note that there are a lot of different `CensusInstance` implementations. They aren't shared between different bot modules. The one in `bots.pr` is only used in two locations, and the other location would probably also like a quick exit on this failure.

Reply:
> 

Yes, I just mean we need to surround a lot of code with try catch in `CheckWorkItem#prRun`.  If it is ok, I will change it.